cmd/server/handlers/handlers.go

```
@@ -73,9 +76,10 @@ func (ah AsyncHandler) GetValue(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusBadRequest)
 	}
 
+	ctx := r.Context()
 	switch m.MType {
 	case agent.GaugeType:
-		v, err := ah.storage.GetGuageByID(m.ID)
+		v, err := ah.storage.GetGuageByID(ctx, m.ID)
 		if err != nil {
 			w.WriteHeader(http.StatusNotFound)
```

Здесь и в подобных местах не хватает return. Ибо значение v после ошибки вообще невалидное

---

early exit или fail fast называется